doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Samsung-Artik-5/index.md

```diff
@@ -50,71 +65,68 @@ Install the third-party components as follows.
+        wget https://raw.githubusercontent.com/kaaproject/kaa/master/client/client-multi/client-cpp/toolchains/rpi.cmake
+        wget https://raw.githubusercontent.com/kaaproject/kaa/develop/client/client-multi/client-cpp/tools/avro-cpp-disable-tests.patch
         patch < avro-cpp-disable-tests.patch
         mkdir build && cd build
-        cmake -DCMAKE_INSTALL_PREFIX:PATH=/opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr -DCMAKE_TOOLCHAIN_FILE=../artik.cmake ..
+        cmake -DCMAKE_INSTALL_PREFIX:PATH=$INSTALLDIR -DCMAKE_TOOLCHAIN_FILE=../rpi.cmake ..
         make && make install
 
 1. Install Botan.
 
+        cd $ARTIK_ROOT
         wget http://botan.randombit.net/releases/Botan-1.11.28.tgz
         tar -xvzf Botan-1.11.28.tgz
         cd Botan-1.11.28
         export PATH=$PATH:/opt/gcc-linaro-arm-linux-gnueabihf/bin
-        python configure.py --cpu=arm --cc-bin=arm-linux-gnueabihf-g++ --prefix=/opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr
+        python configure.py --cpu=arm --cc-bin=arm-linux-gnueabihf-g++ --prefix=$INSTALLDIR
         make install
-        mv /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/include/botan-1.11/botan/ /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/include
-        rm -r /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/include/botan-1.11
+        mv $INSTALLDIR/include/botan-1.11/botan/ $INSTALLDIR/include
+        rm -r $INSTALLDIR/include/botan-1.11
 
 1. Copy libraries to the Artik device.
 
-        scp /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/lib/libboost_*.so* root@<put Artik ip address here>:/usr/lib
-        scp /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/lib/libbotan*.so* root@<put Artik ip address here>:/usr/lib
-        scp /opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/lib/libavrocpp*.so* root@<put Artik ip address here>:/usr/lib
+        scp /opt/gcc-linaro-arm-linux-gnueabihf/lib/libboost_*.so* root@<put Artik ip address here>:/usr/lib
+        scp /opt/gcc-linaro-arm-linux-gnueabihf/lib/libbotan*.so* root@<put Artik ip address here>:/usr/lib
+        scp /opt/gcc-linaro-arm-linux-gnueabihf/lib/libavrocpp*.so* root@<put Artik ip address here>:/usr/lib
 
 # Create application
 
 Now, dependencies are built and it is time to create Kaa application.
 Artik board is running regular Linux, so you can refer to [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/#c-sdk-build) for detailed process of application creation.
 But remember, you must specify [the CMake toolchain file](http://www.vtk.org/Wiki/CMake_Cross_Compiling) when compiling your Kaa application for Artik:
 
-        cmake -DKAA_MAX_LOG_LEVEL=3 -DCMAKE_TOOLCHAIN_FILE=../artik.cmake ..
+        cmake -DKAA_MAX_LOG_LEVEL=3 -DCMAKE_TOOLCHAIN_FILE=../toolchains/rpi.cmake ..
         make
```

Comment:
This file should be renamed into something like `arm-linux.cmake` and should be used for such targets.
However, this probably should be done in scope of another ticket.